allauth/socialaccount/providers/base.py

```diff
@@ -24,6 +24,17 @@ class AuthError(object):
 
 
 class Provider(object):
+    alt_text = None
+    display_name = None
+    name = None
```

Comment:
I'm not sure if there is a standard way to validate required attributes, but this seems reasonable.

`AttributeError` may be slightly more appropriate, but I'm not sure.

Part of me isn't a fan of extra code to make this kind of thing a little bit more explicit (since an exception would happen even without the extra check and would only be encountered by people either doing very custom things, or helping to add a provider to django-allauth itself), but it's quite possible that I'd be in the minority of Python/Django programmers in that regard. @pennersr may have an opinion for how things should work in django-allauth.

My main concern with the original pull request is that it would result in visual oddities but no explicit exceptions if someone left off the `name` attribute in a new provider (though actually, looking at the existing code, I'm not sure anything actually called `self.name` from Python, so the template would have probably seen a `None` anyways).